google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/PublisherImpl.java

```diff
@@ -209,12 +210,12 @@ private void processBatch(Collection<UnbatchedMessage> batch) throws StatusExcep
       return ApiFutures.immediateFailedFuture(error.asException());
     }
     try (CloseableMonitor.Hold h = monitor.enter()) {
-      if (shutdown) {
-        return ApiFutures.immediateFailedFuture(
-            Status.FAILED_PRECONDITION
-                .withDescription("Published after the stream shut down.")
-                .asException());
-      }
+      ApiService.State currentState = state();
+      checkState(
+          currentState == ApiService.State.RUNNING,
+          String.format("Cannot publish when Publisher state is %s.", currentState.name()));
+      // This is an assertion. If the service is running, it should not be shutdown.
```

Comment:
Please remove this comment. This is redundant.

Reply:
Done